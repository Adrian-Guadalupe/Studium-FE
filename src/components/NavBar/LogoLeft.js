import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";
import { ContainerDiv, SynapsBrain } from "..";
import { ReactComponent as SmallWhiteLogo } from "../../images/SmallWhiteLogo.svg";
import { useAppHooks, sizes } from "../../customHooks/useAppHooks.js";
import { SynapsLogoText, SynapsFavicon } from "../../svgComponents";

/**
 *  LogoLeft
 *
 *  @component
 *
 */
const LogoLeft = () => {
  const { theme, pathname } = useAppHooks();
  
  if( theme.screenWidth > sizes.tablet ){
    return (
      <ContainerDiv height={ "100%" } position={ "relative" }
                    width={ "250px" } flexDirection={ "row" }
                    justifyContent={ "flex-start" }>
        <SynapsFavicon height={ "100%" } width={ "40%" }/>
        <SynapsLogoText fill={ "#36405C" } width={ "60%" } height={ "100%" }/>
      </ContainerDiv>
    );
  }else{
    return (
      <ContainerDiv>
        <SynapsLogoText fill={ "#36405C" }/>
      </ContainerDiv>
    );
  }
  
};

LogoLeft.propTypes = {};

export default LogoLeft;