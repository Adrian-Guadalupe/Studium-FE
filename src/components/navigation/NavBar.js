import React from "react";
import styled from "styled-components";
import ButtonTop from "../common/ButtonTop";
// import {Button} from "../common/Button";
// import {Button} from "../common";
// import Button from "../common/";

const NavWrapper = styled.header`
  /* outline: 1px solid red; */
  box-sizing: border-box;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-content: center;
  /* padding-top: 30px; */
  padding: 20px 20px 0 20px;
  position: sticky;
  top: 0;
  width: 100%;
  /* border: 1px solid #c4c4c4; */
  box-shadow: 0px 5px 5px #c4c4c4;
`;

const Logo = styled.a`
  /* outline: 1px solid red; */
  /* position: absolute; */
  width: 72px;
  height: 26px;
  /* left: 20px; */
  /* top: 32px; */
  font-family: "Poppins", sans-serif;
  font-style: bold;
  font-weight: 500;
  font-size: 17px;
  line-height: 25px;
  color: #2e71fd;
  /* margin-left: 15px; */
`;



export function NavBar() {
  return (
    <NavWrapper>
        <Logo>Studium</Logo>
      {}
      <>
      <ButtonTop type="button">Log In</ButtonTop>
      </>
    </NavWrapper>
  );
}

export default NavBar;
