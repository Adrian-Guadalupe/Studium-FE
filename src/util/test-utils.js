// test-utils.js
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { createGlobalStyle, ThemeProvider } from 'styled-components';
import reset from 'styled-reset';
import { createStore } from 'redux';
import rootReducer from '../reducers';
import React from 'react';

const GlobalStyle = createGlobalStyle`
${reset}
`;

const store = createStore(rootReducer);

const theme = {
  lightLightGray: '#D7D7D7',
  lightGray: '#c4c4c4',
  gray: '#C4C4C4',
  darkGray: '#585858',
  darkDarkGray: '#3a3a3a',

  largeRadius: 14,
  smallRadius: 6,

  navBarTopHeight: 75,
  footerHeight: 50,

  screenHeight: window.innerHeight,
  screenWidth: window.outerWidth,
};

const AllTheProviders = ({ children }) => {
  return (
    <ThemeProvider theme={theme}>
      <Provider store={store}>
        <GlobalStyle />
        {children}
      </Provider>
    </ThemeProvider>
  );
};

/**
 *
 * @param ui
 * @param options
 * @returns {RenderResult}
 */
const customRender = (ui, options) => {
  return render(ui, { wrapper: AllTheProviders, ...options });
};

// re-export everything
export * from '@testing-library/react';

export { customRender as render };
