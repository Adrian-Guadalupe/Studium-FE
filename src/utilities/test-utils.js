import React from 'react';
import { render, queries } from '@testing-library/react';
import { Provider } from 'react-redux';
import { createGlobalStyle, ThemeProvider } from 'styled-components';
import reset from 'styled-reset';
import { applyMiddleware, createStore } from 'redux';
import rootReducer from '../reducers';
import moxios from 'moxios';
import { theme } from './theme.js';
import {
  crashReporter,
  logger,
  rafScheduler,
  readyStatePromise,
  thunk,
  timeoutScheduler,
  vanillaPromise,
} from './customMiddleWare.js';

const GlobalStyle = createGlobalStyle`
${reset}
`;

/**
 * @typedef AppStore store
 * @type {Store}
 */
const store = createStore(
  rootReducer,
  applyMiddleware(
    rafScheduler,
    timeoutScheduler,
    thunk,
    vanillaPromise,
    readyStatePromise,
    logger,
    crashReporter
  )
);

/**
 *
 * @param ui
 * @param options
 * @return {
 *   RenderResults: {RenderResults},
 *   providers: {theme: {ThemeProvider}, store: {Store}},
 *   TestLib: {TestingLibraryReact}
 * }
 *
 */
export const customRender = (ui, options) =>
  render(ui, {
    wrapper: AllTheProviders,
    ...options,
  });

/**
 * Wraps a component in the provides and then returns the wrapped
 * commponent.
 * @param children Reach child components
 * @returns {*}
 * @constructor
 */
const AllTheProviders = ({ children }) => {
  return (
    <ThemeProvider theme={theme}>
      <Provider store={store}>
        <GlobalStyle />
        {children}
      </Provider>
    </ThemeProvider>
  );
};

/**
 *
 * @param {HTMLElement} c
 * @return {[{HTMLElement}]}
 */
export const getChildNodes = c => {
  console.log('inside of get next node.');
  const childNodes = [];
  if (c.hasChildNodes()) {
    c.childNodes.forEach(child => {
      childNodes.push(child);
    });
  }
  return childNodes;
};

/**
 *
 * @param {HTMLElement} c
 * @param {string} type
 * @return {[HTMLElement]}
 */
export const getNodesByType = (c, type) => {
  const stack = [];
  const toReturn = [];
  stack.push(c);
  while (stack.length > 0) {
    let node = stack.pop();
    if (node.hasChildNodes()) {
      node.childNodes.forEach(child => {
        stack.push(child);
      });
    }
    if (node.nodeName === type.toUpperCase()) {
      toReturn.push(node);
    }
  }

  return toReturn;
};


export * from '@testing-library/react';

export { theme, store };
