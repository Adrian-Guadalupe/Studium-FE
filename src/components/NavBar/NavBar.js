import React, { useEffect, useState } from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { ContainerDiv, NavBarAvatar } from "../index.js";
import { ReactComponent as SmallWhiteLogo } from "../../images/SmallWhiteLogo.svg";
import { signOut } from "../../actions";
import theming from "styled-theming";
import {
  useAppHooks, mediaQueries, sizes,
} from "../../customHooks/useAppHooks.js";
import LogoLeft from "./LogoLeft.js";

/**
 * Nav Bar
 *
 * @component
 * @example
 *  return (<NavBar />)
 */
export const NavBar = () => {
  const { usersState, theme, dispatch, changePath, pathname } = useAppHooks();
  const [ menuOpen, setMenuOpen ] = useState( false );
  const [ avatarUrl, setAvatarUrl ] = useState( "" );
  
  useEffect( () => {
    if( usersState.user && usersState.user.photoURL ){
      setAvatarUrl( usersState.user.photoURL );
    }else{
      setAvatarUrl( "" );
    }
  }, [ usersState ] );
  
  const logout = () => {
    setMenuOpen( false );
    signOut( dispatch );
  };
  
  const getSignUpText = () => {
    if( theme.screenWidth > 768 ){
      if( pathname === "/signup" ){
        return <Styledh2>SignIn</Styledh2>;
      }else if( pathname === "/signin" ){
        return <Styledh2>SignUp</Styledh2>;
      }
    }else{
      return false;
    }
  };
  
  return (
    <StyledBar className={ "nav-bar" }>
      <ContainerDiv
        justifyContent={ "space-between" }
        className={ "nav-bar-container" }
        height={ "75px" }
        position={ "relative" }
        overFlowY={ "hidden" }
        backgroundColor={ "transparent" }
      >
        <LogoLeft/>
        
        <NavBarAvatar
          onClick={ logout }
          avatarUrl={ avatarUrl }
          className={ "ant-dropdown-link" }
        />
        
        
        { getSignUpText() }
      </ContainerDiv>
    </StyledBar>
  );
};

NavBar.propTypes = {
  backgroundColor: PropTypes.string,
};

const WhiteLogo = styled( SmallWhiteLogo )`

`;

const backgroundColor = theming( "navStyle", {
  light: props => props.theme.navBarLight,
  dark: props => props.theme.navBarDark,
} );

const StyledBar = styled.div`
  background-color: ${ backgroundColor };
  display: flex;
  justify-content: center;
  z-index: 15;
  position: absolute;
  top: 0;
  width: 100%;
  height: ${ props => props.theme.navBarTopHeight + "px" };

  @media screen and ${ mediaQueries.tablet } {
  
  }
`;

const Styledh2 = styled.h2`
  display: flex;
  align-items: center;
  color: #fff;
  position: absolute;
  width: 95px;
  height: 24px;
  left: 1197px;
  top: 38px;
  font-family: Source Sans Pro;
  font-style: normal;
  font-weight: bold;
  font-size: 28px;
  line-height: 24px;
`;
