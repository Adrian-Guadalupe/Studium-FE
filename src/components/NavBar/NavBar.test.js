import React from 'react';
import {customRender} from '../../utilities/test-utils.js';

import NavBar from './NavBar.js';

test('StyledNavBar render correctly', () => {
  const {container} = customRender(<NavBar />);
  expect(container).toMatchSnapshot()
});
