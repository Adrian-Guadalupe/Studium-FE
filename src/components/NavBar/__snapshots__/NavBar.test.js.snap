// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledNavBar render correctly 1`] = `
.c1 {
  box-sizing: border-box;
  top: 0;
  margin: 0;
  position: relative;
  width: 100vw;
  height: 75px;
  max-height: 100%;
  max-width: 100vw;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  overflow-y: hidden;
}

.c1::-webkit-scrollbar {
  width: 0;
}

.c0 {
  background-color: #0d2545;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  z-index: 15;
  position: absolute;
  top: 0;
  width: 100%;
  height: 74px;
}

@media screen and (min-width:768px) {

}

<div>
  <div
    class="c0 nav-bar"
  >
    <div
      class="c1 nav-bar-container"
      height="75px"
    >
      <svg
        style="position: absolute; left: 6%; top: 50%; transform: transition(0, -53%);"
      >
        SmallWhiteLogo.svg
      </svg>
      <svg
        class="ant-dropdown-link"
        style="position: absolute; top: 15px; right: 8%;"
      >
        ProfileIcons.svg
      </svg>
    </div>
  </div>
</div>
`;
